File: Object-C/BlackFireIM/Example/Lib/MSIMSDK-UIKit/ChatDetail/View/Face/MSMenuCollectionViewCell.m
//
//  MSMenuCollectionViewCell.m
//  BlackFireIM
//
//  Created by benny wang on 2022/1/14.
//

#import "MSMenuCollectionViewCell.h"
#import "MSHeader.h"


@implementation MSMenuCellData


@end

@implementation MSMenuCollectionViewCell

- (instancetype)initWithFrame:(CGRect)frame
{
    if (self = [super initWithFrame:frame]) {
        [self setupViews];
        [self defaultLayout];
    }
    return self;
}

- (void)setupViews
{
    self.backgroundColor = [UIColor d_colorWithColorLight:TMenuCell_Background_Color dark:TMenuCell_Background_Color_Dark];
    _menu = [[UIImageView alloc] init];
    _menu.backgroundColor = [UIColor clearColor];
    [self addSubview:_menu];
}

- (void)defaultLayout
{
}

- (void)setData:(MSMenuCellData *)data
{
    if(data.isSelected){
        self.backgroundColor = [UIColor d_colorWithColorLight:TMenuCell_Selected_Background_Color dark:TMenuCell_Selected_Background_Color_Dark];
        _menu.image = [UIImage imageNamed:data.selectPath];
    }else {
        self.backgroundColor = [UIColor d_colorWithColorLight:TMenuCell_Background_Color dark:TMenuCell_Background_Color_Dark];
        _menu.image = [UIImage imageNamed:data.normalPath];
    }
    //update layout
    CGSize size = self.frame.size;
    _menu.frame = CGRectMake(TMenuCell_Margin, TMenuCell_Margin, size.width - 2 * TMenuCell_Margin, size.height - 2 * TMenuCell_Margin);
    _menu.contentMode = UIViewContentModeScaleAspectFit;

}

@end

File: Swift/MSIM-Swift/Lib/MSIMSDK-UIKit/ChatDetail/View/Face/MSMenuCollectionViewCell.swift
//
//  MSMenuCollectionViewCell.swift
//  MSIM-Swift
//
//  Created by benny wang on 2022/3/7.
//

import UIKit

class MSMenuCollectionViewCell: UICollectionViewCell {
    
    var menu: UIImageView!
    
    var data: MSMenuCellData? {
        didSet {
            if data?.isSelected == true {
                backgroundColor = UIColor.d_color(light: MSMcros.TMenuCell_Selected_Background_Color, dark: MSMcros.TMenuCell_Selected_Background_Color_Dark)
                menu.image = UIImage(named: data?.selectPath ?? "")
            }else {
                backgroundColor = UIColor.d_color(light: MSMcros.TMenuCell_Background_Color, dark: MSMcros.TMenuCell_Background_Color_Dark)
                menu.image = UIImage(named: data?.normalPath ?? "")
            }
            menu.frame = CGRect(x: MSMcros.TMenuCell_Margin, y: MSMcros.TMenuCell_Margin, width: self.width - 2 * MSMcros.TMenuCell_Margin, height: self.height - 2 * MSMcros.TMenuCell_Margin)
            menu.contentMode = .scaleAspectFit
        }
    }
    
    override init(frame: CGRect) {

        super.init(frame: frame)
        
        backgroundColor = UIColor.d_color(light: MSMcros.TMenuCell_Background_Color, dark: MSMcros.TMenuCell_Background_Color_Dark)
        
        menu = UIImageView()
        menu.backgroundColor = .clear
        addSubview(menu)
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}

class MSMenuCellData: NSObject {
    
    var normalPath: String?
    
    var selectPath: String?
    
    var isSelected: Bool = false
}
